spanner/api/Spanner.Samples/UpdateUsingDmlReturningAsyncPostgre.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START spanner_postgresql_update_dml_returning]
+
+using Google.Cloud.Spanner.Data;
+using Google.Cloud.Spanner.V1;
+using System;
+using System.Threading.Tasks;
+
+public class UpdateUsingDmlReturningAsyncPostgreSample
+{
+    public async Task<int> UpdateUsingDmlReturningAsyncPostgre(string projectId, string instanceId, string databaseId)
+    {
+        string connectionString = $"Data Source=projects/{projectId}/instances/{instanceId}/databases/{databaseId}";
+
+        using var connection = new SpannerConnection(connectionString);
+        await connection.OpenAsync();
+
+        //Updates records from Singers table satisfying a particular condition and returns the SingerId
+        //and Rating column of the updated records using ‘RETURNING SingerId, Rating’.
+        //All columns of all the updated records can be returned using ‘RETURNING *’.
+        using var cmd = connection.CreateDmlCommand("UPDATE Singers SET Rating = $1 WHERE SingerId = 12 RETURNING SingerId, Rating", 
+            new SpannerParameterCollection 
+            {
+                { "p1", SpannerDbType.PgNumeric, PgNumeric.Parse("4.0") }
+            });
+
+        var reader = await cmd.ExecuteReaderAsync();
+        var rowCount = 0;
```

Comment:
Though it is fine,  but I will create a `List<Singer>` and return it rather than just row count and writing to console. That we can assert on rowcount as well as data if we want.  That will also be consistent with our `Query...` samples.